core/runtime/src/main/java/io/quarkus/runtime/configuration/RandomPorts.java

```diff
@@ -0,0 +1,27 @@
+package io.quarkus.runtime.configuration;
+
+import static java.net.InetAddress.getByName;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+public class RandomPorts {
+    private static final Map<Integer, Integer> randomPorts = new HashMap<>();
+
+    public static int get(final String host, final int port) {
+        return randomPorts.computeIfAbsent(port, new Function<Integer, Integer>() {
+            @Override
+            public Integer apply(final Integer socketAddress) {
+                try (ServerSocket serverSocket = new ServerSocket(0, 0, getByName(host))) {
+                    serverSocket.setReuseAddress(false);
+                    return serverSocket.getLocalPort();
```

Comment:
We may need to add a grace period here. 
I remember that windows can be tricky. 
(optional, user runtime config)

Reply:
My understanding is that if you set `serverSocket.setReuseAddress(true);` if will allow you to bind it again even if the socket is in timeout: https://docs.oracle.com/javase/8/docs/api/java/net/ServerSocket.html#setReuseAddress-boolean-

Or are you thinking about something else?